include/llama.h

```diff
@@ -1130,6 +1132,8 @@ extern "C" {
                           const char * grammar_str,
                           const char * grammar_root);
 
+    LLAMA_API bool llama_sampler_is_grammar_empty(struct llama_sampler * gsmpl);
```

Comment:
I wonder if instead of extending the `llama_sampler` API, it would be better to pass necessary callbacks (such as `is_empty`, `accept_str`, etc.) through the `llama_sampler_init_grammar()` call.